### ISSUE TITLE
[Feature Request] Ability to sync admins that are in a nested distro for superadmins_group by default

### ISSUE THREAD

**Issue by CoreyGriffin** [google/loaner#71]
### Expected Behaviour

Setting `SUPERADMINS_GROUP` to "group@domain.com" will provide proper permissions to everyone in the group for the GnG web app

### Actual Behaviour

Potentially not an issue but more so an edge case we ran into where we have a bridge group containing two separate google groups  for 2 teams having their own group.  Since both teams require the same permissions, we are setting the parent group as the `SUPERADMINS_GROUP`  the current [get_all_users_in_group](https://github.com/google/loaner/blob/master/loaner/web_app/backend/clients/directory.py#L326)  method does not support nested groups so unless the users are in the top level group...they will not get access

**TLDR**

`admin-group@domain.com`

- `group-team-1@domain.com`
- `group-team-2@domain.com`
- `random-team-member1@domain.com`
- `random-team-member2@domain.com`

Only members of `admin-group@domain.com`  (random-team-member 1 &2) get web app admin permissions. Members of the nested groups do not.

### Steps to Reproduce

Add a nested group to your super admins group and try to perform an admin action in the web app as a member of the nested group

### Workaround

We implemented a workaround for this use case by adjusting the [get_all_users_in_group](https://github.com/google/loaner/blob/master/loaner/web_app/backend/clients/directory.py#L326)  method 

```
def get_all_users_in_group(group_email, users=[]):
	"""Retrieves all of the users in a particular Google Group.
	Args:
		group_email: str, the email address of the group.
	Returns:
		A list of all user's email addresses in the group provided.
	"""
	nested_groups = []
	response = self._users_in_group(group_email)
	
	# There is no next page so build users list and return it
	if not response.get(_NEXT_PAGE):
		members = response.get('members')
		if members:
			for member in members:
				if member.get("type") == "GROUP":
					nested_groups.append(member['email'].lower())
				else:
					users.append(member['email'])
	
		if nested_groups:
			for group in nested_groups[::-1]:
				get_all_users_in_group(group, user_state)
				nested_groups.pop()
				
		return set(users)
		
	# Need to paginate
	while response.get(_NEXT_PAGE):
		for member in response['members']:
			if member.get("type") == "GROUP":
				nested_groups.append(member['email'].lower())
			else:
				users.append(member['email'])
		response = self._users_in_group(group_email, response.get(_NEXT_PAGE))

	
	for member in response['members']:
		if member.get("type") == "GROUP":
			nested_groups.append(member['email'].lower())
		else:
			users.append(member['email'])
	
	if nested_groups:
		for group in nested_groups[::-1]:
			get_all_users_in_group(group, users)
			nested_groups.pop()
			
	return set(users)
```

**Comment by gyoshi02** [comment]
Hey @CoreyGriffin 
I didn't get any error while running the deployment script. 
I can send over any logs you'd like, or any info within cloud console. 
Only thing that I had noticed was that nested groups (which I moved my user account to and from) didn't work still.  
Happy to test any other alternative code updates.

**Comment by CoreyGriffin** [comment]
@gyoshi02 had some time to look at this again today...the reason that script doesn’t work as is was because they are masking the response in constants.py which only pulls the email and pageToken ... you can either add members/type to that mask variable which I believe (not certain) would require a bootstrap or you can just overwrite it in the private _users_in_group method ... I updated the solution above with the changes I made...they _appear_ to have done the trick for me

**Comment by gyoshi02** [comment]
Hey @CoreyGriffin 
Just wanted to check back in here. 
I'm about to run through another deploy.  Pls confirm that I just need to add your amended code to the web_app/backend/clients/directory.py file, or is there a change that must also be made in constants.py?

**Comment by CoreyGriffin** [comment]
@gyoshi02 the code above should work, you may want to double check the indentation if you're just copying and pasting before deploying.  I did not change the constants file...just the code in the directory.py file

**Comment by gyoshi02** [comment]
@CoreyGriffin 
Yup, there was a slight tab issue when I copied, but was able to get it working after a couple of re-deploys.  I think our nested groups are working now, but will report back, after testing further tomorrow.

**Comment by CoreyGriffin** [comment]
@gyoshi02 everything end up working for you?

**Comment by gyoshi02** [comment]
@coreyGriffin 
I’m all good with my nested google groups. 
Was able to confirm the users that should have access based on the groups. 
Thanks for quick response, and follow up.
